### PR TITLE
Stop assuming deps of go_proto_library has importmap attribute

### PR DESCRIPTION
**Pull request by linzhp**
**What type of PR is this?**

Bug fix

**What does this PR do? Why is it needed?**
`_go_proto_aspect` applies to all `deps` of `go_proto_library` transitively, following the `deps` and `embed` attributes. Meanwhile, the only requirement for `deps` of `go_proto_library` is to provide `GoInfo`. They don't have to be `go_proto_library` or `go_library`. We should not assume they also have `importmap`, `importpath_aliases` or `embed` attributes. They are likely to have `importpath`, otherwise Gazelle won't be able to resolve them. So we can keep the assumption of `importpath` for now until breakages are discovered.

cc @dzbarsky

### CODE REVIEW

File: proto/def.bzl [hunk 1]
```diff
@@ -74,9 +74,6 @@ def _go_proto_aspect_impl(_target, ctx):
         attr,
         include_deprecated_properties = False,
         importpath = attr.importpath,
-        importmap = attr.importmap,
```
Comment (fmeum) [inline proto/def.bzl:77]: I would have expected logic such as `importmap = getattr(attr, "importmap", None)`. Is it clear that we don't need this?

**Comment by dzbarsky** [comment]
I didn't think people would add custom rule deps to go_proto_library. What's the use case?

File: proto/def.bzl (hunk 1 above)
Comment (linzhp) [reply proto/def.bzl:77]: `go_context` calls `getattr` automatically if importmap etc are not passed: https://github.com/bazel-contrib/rules_go/blob/702d10674ec48a679488a725c86bf35121420698/go/private/context.bzl#L581

**Review by fmeum — APPROVED** [review]

**Comment by linzhp** [comment]
> I didn't think people would add custom rule deps to go_proto_library. What's the use case?

There could be custom protoc plugins that add dependency to custom generated code

**Comment by dzbarsky** [comment]
> > I didn't think people would add custom rule deps to go_proto_library. What's the use case?
> 
> There could be custom protoc plugins that add dependency to custom generated code

I see, and the custom generated code isn't a normal go_library? Fair enough. Perhaps it would be nice to add a testcase for this usage to the repo to avoid it breaking in the future, since I suspect this is not a very common setup among rules_go users.

**Pull request merged**